app/public/js/profile.js

```diff
@@ -4,41 +4,18 @@ profile.js is used in the profile.blade.php view.
 
 var regions = [];
 
-/**
-When select-all checkboxes are checked, check every other associated checkbox.
-*/
-function selectAllChanged() {
-	if ( $(this).is(':checked') ) {
-		var tab = $(this).closest('[role="tabpanel"]');
-		var questions = tab.find('.questions input[type="checkbox"]');
-		questions.prop('checked', true);
-	}
-}
-
-function updateSelectAll() {
-	var tab = $(this).closest('[role="tabpanel"]');
-	var selectAll = tab.find('input.select-all');
-	var checkboxes = tab.find('.questions input[type="checkbox"]');
-	var allChecked = true;
-	checkboxes.each(function() {
-		if ( !$(this).is(':checked') )
-			allChecked = false;
-	});
-	selectAll.prop('checked', allChecked);
-}
-
-function initSelectAllBindings() {
-	var interests_box = $('.accessibility-interests');
-	var select_all_elements = interests_box.find('input.select-all');
-	var checkboxes = interests_box.find('.questions input');
-	var question_categories = interests_box.find('.category > .checkbox');
-	select_all_elements.change(selectAllChanged);
-	checkboxes.change(updateSelectAll);
-	question_categories.each(function() {
-		updateSelectAll.call(this);
-	});
+function selectAllToggle() {
```

Comment:
The button isn't updating properly when the associated checkboxes are updated.

I went through these steps and it didn't work as desired:
1. Click "Select All".  All checkboxes are checked and the button changes to say "Unselect All".  This is good.
1. Uncheck one of the associated checkboxes.  The "Unselect All" stays as is.  This is not good.  It should change back to "Select All".

Here is a screenshot showing the undesired state:
![image](https://cloud.githubusercontent.com/assets/11838191/26039136/cf7171ba-38e4-11e7-85aa-e9cae0e6fab7.png)